packages/cysync-core/src/pages/MainApp/Settings/Tabs/AppSettings.tsx

```diff
@@ -20,6 +21,25 @@ export const AppSettings: React.FC = () => {
   const [isAutoUpdateCySyncEnabled, setAutoUpdateCySyncEnabled] =
     useState<boolean>(false);
 
+  useEffect(() => {
+    keyValueStore.isAnalyticsAndBugReportEnabled
+      .get()
+      .then(setAnalyticsAndBugReportEnabled);
+    keyValueStore.isAutoUpdateCySyncEnabled
+      .get()
+      .then(setAutoUpdateCySyncEnabled);
+  }, []);
+
+  useEffect(() => {
+    keyValueStore.isAnalyticsAndBugReportEnabled.set(
+      isAnalyticsAndBugReportEnabled,
+    );
+  }, [isAnalyticsAndBugReportEnabled]);
+
+  useEffect(() => {
+    keyValueStore.isAutoUpdateCySyncEnabled.set(isAutoUpdateCySyncEnabled);
+  }, [isAutoUpdateCySyncEnabled]);
```

Comment:
Keep the `useState` type as `boolean | undefined`. And if it is `undefined`, it means it hasn't been set yet by the  `.get()` function, in which case we don't set the variable on DB.

Reply:
Implemented and added loading state to `Toggle` Component